services/prescriptionService.js

```diff
@@ -60,11 +60,17 @@ const deletePrescription = async (prescriptionId) => {
 
 const completePartialPrescription = async (
   prescriptionId,
-  updatedMedicinesIds
+  updatedMedicinesIds,
+  pharmacistId
 ) => {
   const prescription = await Prescription.findById(prescriptionId).populate(
     "medicines"
   );
+  const pharmacist = await User.findById(pharmacistId);
+
+  if (!pharmacist || !(pharmacist.role === "pharmacist")) {
```

Comment:
If you want to check if the current role is different than "pharmacist" you must preform the check like this
`if(!pharmacist || pharmacist.role !== "pharmacist")`